### PR TITLE
Edit user profile

### PR DESCRIPTION
**Pull request by Winston-Hsiao**
**What does this PR do?**

Allow logged in users to edit their profile when on their profile page.

**What issues does this PR fix or reference?**

#292 

** UI Changes **
Screenshot 2024-08-27 at 6 00 26 PM
Screenshot 2024-08-27 at 6 01 44 PM

### CODE REVIEW

File: store/app/routers/users.py [hunk 1]
```diff
@@ -65,46 +65,39 @@ async def maybe_get_request_api_key_id(request: Request) -> str | None:
...
+    api_key_id: str = Depends(get_request_api_key_id),
+) -> User:
+    return await get_session_user_with_permission("read", crud, api_key_id)
+
+
 async def get_session_user_with_write_permission(
-    crud: Annotated[Crud, Depends(Crud.get)],
-    api_key_id: Annotated[str, Depends(get_request_api_key_id)],
+    crud: Crud = Depends(Crud.get),
+    api_key_id: str = Depends(get_request_api_key_id),
```
Comment (codekansas) [inline store/app/routers/users.py:91]: why this change? i think `Annotated[...]` is preferable

Comment (Winston-Hsiao) [reply store/app/routers/users.py:91]: Ah I was experimenting with fixing a weird linting issue, I'll put that back. I've still got some work to do with this to get it to where it needs to be.

File: store/app/crud/users.py [hunk 2]
```diff
@@ -156,6 +158,31 @@ async def delete_api_key(self, token: APIKey | str) -> None:
...
+
+        update_dict = {}
+        for key, value in updates.items():
+            if hasattr(user, key):
+                update_dict[key] = value
+            else:
+                raise ValueError(f"Invalid field: {key}")
+
+        user.update_timestamp()
+        update_dict["updated_at"] = user.updated_at
```
Comment (codekansas) [inline store/app/crud/users.py:175]: one small question, isn't this logic handled in `_update_item` already?

**Review by codekansas — APPROVED** [review]
lgtm!

File: store/app/crud/users.py (hunk 2 above)
Comment (Winston-Hsiao) [reply store/app/crud/users.py:175]: ah yeah it is redundant. Cleaning up `update_user` and then will merge.
